File: apps/dsa-web/src/api/adminOpsStatus.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export interface AdminOpsStatusSection {
  available: boolean;
  status: string;
  label: string;
  reasonCode?: string | null;
  readOnly: boolean;
  noExternalCalls: boolean;
  advisoryOnly: boolean;
  liveEnforcement: boolean;
  enforcementEnabled: boolean;
  runtimeBehaviorChanged: boolean;
  consumerVisible: boolean;
  providerBehaviorChanged: boolean;
  marketCacheBehaviorChanged: boolean;
  deleteAllowed: boolean;
  dataSources: string[];
  summary: Record<string, unknown>;
  limitations: string[];
}

export interface AdminOpsAdvisoryVsEnforcement {
  label: string;
  enforcementLabel: string;
  sourceUnavailableBehavior: string;
  readOnly: boolean;
  noExternalCalls: boolean;
  liveEnforcement: boolean;
  runtimeBehaviorChanged: boolean;
  consumerVisible: boolean;
}

export interface AdminOpsCockpitFollowUpProposal {
  proposalKey: string;
  title: string;
  approvalNeeded: boolean;
  likelyFiles: string[];
  risk: string;
  validation: string[];
}

export interface AdminOpsCockpitDomain {
  domainKey: string;
  label: string;
  status: string;
  statusLabel: string;
  detailRoute: string;
  foundationLanded: boolean;
  evidenceToolingPresent: boolean;
  realOperatorEvidenceMissing: boolean;
  approvalRequired: boolean;
  publicLaunchNoGo: boolean;
  readOnly: boolean;
  advisoryOnly: boolean;
  noExternalCalls: boolean;
  liveEnforcement: boolean;
  runtimeBehaviorChanged: boolean;
  providerRuntimeChanged: boolean;
  externalActionsEnabled: boolean;
  evidenceRefs: string[];
  blockerRefs: string[];
  safeNextActions: string[];
  limitations: string[];
  priorityRank: number;
  priorityTier: string;
  impactLevel: string;
  recommendedNextAction: string;
  blockingReasonSummary: string;
  ownerSurface: string;
  remediationSurface: string;
  followUpProposals: AdminOpsCockpitFollowUpProposal[];
}

export interface AdminOpsCockpitBlocker {
  blockerKey: string;
  title: string;
  severity: string;
  publicLaunchNoGo: boolean;
  approvalRequired: boolean;
  affectedDomains: string[];
  evidenceRefs: string[];
  nextAction: string;
}

export interface AdminOpsCockpitMaintenanceQueueItem {
  domainKey: string;
  label: string;
  status: string;
  priorityRank: number;
  priorityTier: string;
  impactLevel: string;
  recommendedNextAction: string;
  blockingReasonSummary: string;
  ownerSurface: string;
  remediationSurface: string;
}

export interface AdminOpsLaunchCockpit {
  contract: string;
  readOnly: boolean;
  advisoryOnly: boolean;
  noExternalCalls: boolean;
  publicLaunchApproved: boolean;
  publicLaunchNoGo: boolean;
  liveEnforcement: boolean;
  runtimeBehaviorChanged: boolean;
  approvalRequired: boolean;
  summaryCounts: Record<string, number>;
  unsafeActionStates: Record<string, boolean>;
  domains: AdminOpsCockpitDomain[];
  recommendedMaintenanceQueue: AdminOpsCockpitMaintenanceQueueItem[];
  blockers: AdminOpsCockpitBlocker[];
  safeNextActions: string[];
  limitations: string[];
  prioritySummary: Record<string, number>;
}

export interface AdminOpsStatusResponse {
  generatedAt: string;
  readOnly: boolean;
  noExternalCalls: boolean;
  liveEnforcement: boolean;
  runtimeBehaviorChanged: boolean;
  consumerVisible: boolean;
  advisoryVsEnforcement: AdminOpsAdvisoryVsEnforcement;
  providerStatusSummary?: AdminOpsStatusSection;
  quotaCostAdvisoryStatusSummary?: AdminOpsStatusSection;
  storageReadinessSummary?: AdminOpsStatusSection;
  taskQueueStatusSummary?: AdminOpsStatusSection;
  adminLogEvidenceSummary?: AdminOpsStatusSection;
  launchCockpit: AdminOpsLaunchCockpit;
  metadata?: Record<string, unknown>;
}

export interface AdminScannerUniverseReadinessResponse {
  contractVersion: string;
  status: string;
  scannerUniverseStatus?: string | null;
  market: string;
  profile: string;
  freshnessState: string;
  lastUpdatedAt?: string | null;
  universeSize: number;
  affectedProductSurfaces: string[];
  nextOperatorAction: string;
  scannerUniverseReadiness: Record<string, unknown>;
  candidateGenerationState?: string | null;
  candidateGenerationBlockers: string[];
  readOnly: boolean;
  noExternalCalls: boolean;
  mutationEnabled: boolean;
  providerCallsEnabled: boolean;
  consumerVisible: boolean;
}

export interface AdminScannerUniverseRefreshResponse {
  contractVersion: string;
  status: string;
  actionStatus: string;
  market: string;
  profile: string;
  refreshExecuted: boolean;
  mutationEnabled: boolean;
  noExternalCalls: boolean;
  providerCallsEnabled: boolean;
  runtimeBehaviorChanged: boolean;
  nextOperatorAction: string;
  before: Record<string, unknown>;
  after: Record<string, unknown>;
}

function arrayOfStrings(value: unknown): string[] {
  return Array.isArray(value)
    ? value.flatMap((item) => {
      const text = String(item || '').trim();
      return text ? [text] : [];
    })
    : [];
}

function normalizeProposal(payload: Record<string, unknown>): AdminOpsCockpitFollowUpProposal {
  const normalized = toCamelCase<AdminOpsCockpitFollowUpProposal>(payload);
  return {
    proposalKey: String(normalized.proposalKey || ''),
    title: String(normalized.title || ''),
    approvalNeeded: normalized.approvalNeeded !== false,
    likelyFiles: arrayOfStrings(normalized.likelyFiles),
    risk: String(normalized.risk || ''),
    validation: arrayOfStrings(normalized.validation),
  };
}

function normalizeDomain(payload: Record<string, unknown>): AdminOpsCockpitDomain {
  const normalized = toCamelCase<AdminOpsCockpitDomain>(payload);
  return {
    domainKey: String(normalized.domainKey || ''),
    label: String(normalized.label || ''),
    status: String(normalized.status || ''),
    statusLabel: String(normalized.statusLabel || ''),
    detailRoute: String(normalized.detailRoute || '/settings/system'),
    foundationLanded: Boolean(normalized.foundationLanded),
    evidenceToolingPresent: Boolean(normalized.evidenceToolingPresent),
    realOperatorEvidenceMissing: Boolean(normalized.realOperatorEvidenceMissing),
    approvalRequired: normalized.approvalRequired !== false,
    publicLaunchNoGo: normalized.publicLaunchNoGo !== false,
    readOnly: normalized.readOnly !== false,
    advisoryOnly: normalized.advisoryOnly !== false,
    noExternalCalls: normalized.noExternalCalls !== false,
    liveEnforcement: Boolean(normalized.liveEnforcement),
    runtimeBehaviorChanged: Boolean(normalized.runtimeBehaviorChanged),
    providerRuntimeChanged: Boolean(normalized.providerRuntimeChanged),
    externalActionsEnabled: Boolean(normalized.externalActionsEnabled),
    evidenceRefs: arrayOfStrings(normalized.evidenceRefs),
    blockerRefs: arrayOfStrings(normalized.blockerRefs),
    safeNextActions: arrayOfStrings(normalized.safeNextActions),
    limitations: arrayOfStrings(normalized.limitations),
    priorityRank: Number(normalized.priorityRank || 0),
    priorityTier: String(normalized.priorityTier || 'watch'),
    impactLevel: String(normalized.impactLevel || 'low'),
    recommendedNextAction: String(normalized.recommendedNextAction || ''),
    blockingReasonSummary: String(normalized.blockingReasonSummary || ''),
    ownerSurface: String(normalized.ownerSurface || 'admin_maintenance'),
    remediationSurface: String(normalized.remediationSurface || normalized.detailRoute || '/admin'),
    followUpProposals: Array.isArray(normalized.followUpProposals)
      ? normalized.followUpProposals.map((item) => normalizeProposal(item as unknown as Record<string, unknown>))
      : [],
  };
}

function normalizeQueueItem(payload: Record<string, unknown>): AdminOpsCockpitMaintenanceQueueItem {
  const normalized = toCamelCase<AdminOpsCockpitMaintenanceQueueItem>(payload);
  return {
    domainKey: String(normalized.domainKey || ''),
    label: String(normalized.label || ''),
    status: String(normalized.status || ''),
    priorityRank: Number(normalized.priorityRank || 0),
    priorityTier: String(normalized.priorityTier || 'watch'),
    impactLevel: String(normalized.impactLevel || 'low'),
    recommendedNextAction: String(normalized.recommendedNextAction || ''),
    blockingReasonSummary: String(normalized.blockingReasonSummary || ''),
    ownerSurface: String(normalized.ownerSurface || 'admin_maintenance'),
    remediationSurface: String(normalized.remediationSurface || '/admin'),
  };
}

function normalizeBlocker(payload: Record<string, unknown>): AdminOpsCockpitBlocker {
  const normalized = toCamelCase<AdminOpsCockpitBlocker>(payload);
  return {
    blockerKey: String(normalized.blockerKey || ''),
    title: String(normalized.title || ''),
    severity: String(normalized.severity || 'high'),
    publicLaunchNoGo: normalized.publicLaunchNoGo !== false,
    approvalRequired: normalized.approvalRequired !== false,
    affectedDomains: arrayOfStrings(normalized.affectedDomains),
    evidenceRefs: arrayOfStrings(normalized.evidenceRefs),
    nextAction: String(normalized.nextAction || ''),
  };
}

function normalizeLaunchCockpit(payload: Record<string, unknown> | undefined): AdminOpsLaunchCockpit {
  const normalized = toCamelCase<AdminOpsLaunchCockpit>(payload || {});
  return {
    contract: String(normalized.contract || 'admin_ops_launch_cockpit_v1'),
    readOnly: normalized.readOnly !== false,
    advisoryOnly: normalized.advisoryOnly !== false,
    noExternalCalls: normalized.noExternalCalls !== false,
    publicLaunchApproved: Boolean(normalized.publicLaunchApproved),
    publicLaunchNoGo: normalized.publicLaunchNoGo !== false,
    liveEnforcement: Boolean(normalized.liveEnforcement),
    runtimeBehaviorChanged: Boolean(normalized.runtimeBehaviorChanged),
    approvalRequired: normalized.approvalRequired !== false,
    summaryCounts: normalized.summaryCounts && typeof normalized.summaryCounts === 'object'
      ? Object.fromEntries(Object.entries(normalized.summaryCounts).map(([key, value]) => [key, Number(value || 0)]))
      : {},
    unsafeActionStates: normalized.unsafeActionStates && typeof normalized.unsafeActionStates === 'object'
      ? Object.fromEntries(Object.entries(normalized.unsafeActionStates).map(([key, value]) => [key, Boolean(value)]))
      : {},
    domains: Array.isArray(normalized.domains)
      ? normalized.domains.map((item) => normalizeDomain(item as unknown as Record<string, unknown>))
      : [],
    recommendedMaintenanceQueue: Array.isArray(normalized.recommendedMaintenanceQueue)
      ? normalized.recommendedMaintenanceQueue.map((item) => normalizeQueueItem(item as unknown as Record<string, unknown>))
      : [],
    blockers: Array.isArray(normalized.blockers)
      ? normalized.blockers.map((item) => normalizeBlocker(item as unknown as Record<string, unknown>))
      : [],
    safeNextActions: arrayOfStrings(normalized.safeNextActions),
    limitations: arrayOfStrings(normalized.limitations),
    prioritySummary: normalized.prioritySummary && typeof normalized.prioritySummary === 'object'
      ? Object.fromEntries(Object.entries(normalized.prioritySummary).map(([key, value]) => [key, Number(value || 0)]))
      : {},
  };
}

function normalizeScannerUniverseReadiness(
  payload: Record<string, unknown>,
): AdminScannerUniverseReadinessResponse {
  const normalized = toCamelCase<AdminScannerUniverseReadinessResponse>(payload || {});
  return {
    contractVersion: String(normalized.contractVersion || 'scanner_universe_operator_readiness_v1'),
    status: String(normalized.status || 'unavailable'),
    scannerUniverseStatus: normalized.scannerUniverseStatus ? String(normalized.scannerUniverseStatus) : null,
    market: String(normalized.market || ''),
    profile: String(normalized.profile || ''),
    freshnessState: String(normalized.freshnessState || 'unknown'),
    lastUpdatedAt: normalized.lastUpdatedAt ? String(normalized.lastUpdatedAt) : null,
    universeSize: Number(normalized.universeSize || 0),
    affectedProductSurfaces: arrayOfStrings(normalized.affectedProductSurfaces),
    nextOperatorAction: String(normalized.nextOperatorAction || ''),
    scannerUniverseReadiness: normalized.scannerUniverseReadiness && typeof normalized.scannerUniverseReadiness === 'object'
      ? normalized.scannerUniverseReadiness as Record<string, unknown>
      : {},
    candidateGenerationState: normalized.candidateGenerationState ? String(normalized.candidateGenerationState) : null,
    candidateGenerationBlockers: arrayOfStrings(normalized.candidateGenerationBlockers),
    readOnly: normalized.readOnly !== false,
    noExternalCalls: normalized.noExternalCalls !== false,
    mutationEnabled: Boolean(normalized.mutationEnabled),
    providerCallsEnabled: Boolean(normalized.providerCallsEnabled),
    consumerVisible: Boolean(normalized.consumerVisible),
  };
}

function normalizeScannerUniverseRefresh(
  payload: Record<string, unknown>,
): AdminScannerUniverseRefreshResponse {
  const normalized = toCamelCase<AdminScannerUniverseRefreshResponse>(payload || {});
  return {
    contractVersion: String(normalized.contractVersion || 'scanner_universe_operator_action_v1'),
    status: String(normalized.status || 'manual_action_required'),
    actionStatus: String(normalized.actionStatus || 'deferred'),
    market: String(normalized.market || ''),
    profile: String(normalized.profile || ''),
    refreshExecuted: Boolean(normalized.refreshExecuted),
    mutationEnabled: Boolean(normalized.mutationEnabled),
    noExternalCalls: normalized.noExternalCalls !== false,
    providerCallsEnabled: Boolean(normalized.providerCallsEnabled),
    runtimeBehaviorChanged: Boolean(normalized.runtimeBehaviorChanged),
    nextOperatorAction: String(normalized.nextOperatorAction || ''),
    before: normalized.before && typeof normalized.before === 'object'
      ? normalized.before as Record<string, unknown>
      : {},
    after: normalized.after && typeof normalized.after === 'object'
      ? normalized.after as Record<string, unknown>
      : {},
  };
}

export const adminOpsStatusApi = {
  async getStatus(): Promise<AdminOpsStatusResponse> {
    const response = await apiClient.get<Record<string, unknown>>('/api/v1/admin/ops/status');
    const normalized = toCamelCase<AdminOpsStatusResponse>(response.data || {});
    return {
      ...normalized,
      readOnly: normalized.readOnly !== false,
      noExternalCalls: normalized.noExternalCalls !== false,
      liveEnforcement: Boolean(normalized.liveEnforcement),
      runtimeBehaviorChanged: Boolean(normalized.runtimeBehaviorChanged),
      consumerVisible: Boolean(normalized.consumerVisible),
      launchCockpit: normalizeLaunchCockpit(normalized.launchCockpit as unknown as Record<string, unknown>),
    };
  },
  async getScannerUniverseReadiness(market: 'us' | 'cn'): Promise<AdminScannerUniverseReadinessResponse> {
    const response = await apiClient.get<Record<string, unknown>>(
      `/api/v1/admin/ops/scanner-universe-readiness?market=${encodeURIComponent(market)}`,
    );
    return normalizeScannerUniverseReadiness(response.data || {});
  },
  async requestScannerUniverseRefresh(market: 'us' | 'cn'): Promise<AdminScannerUniverseRefreshResponse> {
    const response = await apiClient.post<Record<string, unknown>>(
      `/api/v1/admin/ops/scanner-universe-refresh?market=${encodeURIComponent(market)}`,
    );
    return normalizeScannerUniverseRefresh(response.data || {});
  },
};

File: api/v1/schemas/admin_ops_status.py
# -*- coding: utf-8 -*-
"""Schemas for read-only admin ops status snapshots."""

from __future__ import annotations

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, ConfigDict, Field


class _AdminOpsStatusModel(BaseModel):
    model_config = ConfigDict(populate_by_name=True)


class AdminOpsStatusSection(_AdminOpsStatusModel):
    available: bool = False
    status: str = "unavailable"
    service: str = ""
    configured: bool = False
    last_checked_at: Optional[str] = Field(default=None, alias="lastCheckedAt")
    message: str = ""
    label: str = "advisory"
    reason_code: Optional[str] = Field(default=None, alias="reasonCode")
    read_only: bool = Field(default=True, alias="readOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    advisory_only: bool = Field(default=True, alias="advisoryOnly")
    live_enforcement: bool = Field(default=False, alias="liveEnforcement")
    enforcement_enabled: bool = Field(default=False, alias="enforcementEnabled")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    consumer_visible: bool = Field(default=False, alias="consumerVisible")
    provider_behavior_changed: bool = Field(default=False, alias="providerBehaviorChanged")
    market_cache_behavior_changed: bool = Field(default=False, alias="marketCacheBehaviorChanged")
    delete_allowed: bool = Field(default=False, alias="deleteAllowed")
    data_sources: List[str] = Field(default_factory=list, alias="dataSources")
    summary: Dict[str, Any] = Field(default_factory=dict)
    limitations: List[str] = Field(default_factory=list)


class AdminOpsAdvisoryVsEnforcement(_AdminOpsStatusModel):
    label: str = "advisory_snapshot"
    enforcement_label: str = Field(default="not_launch_control", alias="enforcementLabel")
    source_unavailable_behavior: str = Field(default="degrade_to_unavailable", alias="sourceUnavailableBehavior")
    read_only: bool = Field(default=True, alias="readOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    live_enforcement: bool = Field(default=False, alias="liveEnforcement")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    consumer_visible: bool = Field(default=False, alias="consumerVisible")


class AdminOpsCockpitFollowUpProposal(_AdminOpsStatusModel):
    proposal_key: str = Field(alias="proposalKey")
    title: str
    approval_needed: bool = Field(default=True, alias="approvalNeeded")
    likely_files: List[str] = Field(default_factory=list, alias="likelyFiles")
    risk: str
    validation: List[str] = Field(default_factory=list)


class AdminOpsCockpitDomain(_AdminOpsStatusModel):
    domain_key: str = Field(alias="domainKey")
    label: str
    status: str
    status_label: str = Field(alias="statusLabel")
    detail_route: str = Field(alias="detailRoute")
    foundation_landed: bool = Field(alias="foundationLanded")
    evidence_tooling_present: bool = Field(alias="evidenceToolingPresent")
    real_operator_evidence_missing: bool = Field(alias="realOperatorEvidenceMissing")
    approval_required: bool = Field(default=True, alias="approvalRequired")
    public_launch_no_go: bool = Field(default=True, alias="publicLaunchNoGo")
    read_only: bool = Field(default=True, alias="readOnly")
    advisory_only: bool = Field(default=True, alias="advisoryOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    live_enforcement: bool = Field(default=False, alias="liveEnforcement")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    provider_runtime_changed: bool = Field(default=False, alias="providerRuntimeChanged")
    external_actions_enabled: bool = Field(default=False, alias="externalActionsEnabled")
    evidence_refs: List[str] = Field(default_factory=list, alias="evidenceRefs")
    blocker_refs: List[str] = Field(default_factory=list, alias="blockerRefs")
    safe_next_actions: List[str] = Field(default_factory=list, alias="safeNextActions")
    limitations: List[str] = Field(default_factory=list)
    priority_rank: int = Field(default=0, alias="priorityRank")
    priority_tier: str = Field(default="watch", alias="priorityTier")
    impact_level: str = Field(default="low", alias="impactLevel")
    recommended_next_action: str = Field(default="", alias="recommendedNextAction")
    blocking_reason_summary: str = Field(default="", alias="blockingReasonSummary")
    owner_surface: str = Field(default="admin_maintenance", alias="ownerSurface")
    remediation_surface: str = Field(default="/admin", alias="remediationSurface")
    follow_up_proposals: List[AdminOpsCockpitFollowUpProposal] = Field(
        default_factory=list,
        alias="followUpProposals",
    )


class AdminOpsCockpitBlocker(_AdminOpsStatusModel):
    blocker_key: str = Field(alias="blockerKey")
    title: str
    severity: str
    public_launch_no_go: bool = Field(default=True, alias="publicLaunchNoGo")
    approval_required: bool = Field(default=True, alias="approvalRequired")
    affected_domains: List[str] = Field(default_factory=list, alias="affectedDomains")
    evidence_refs: List[str] = Field(default_factory=list, alias="evidenceRefs")
    next_action: str = Field(alias="nextAction")


class AdminOpsCockpitMaintenanceQueueItem(_AdminOpsStatusModel):
    domain_key: str = Field(alias="domainKey")
    label: str
    status: str
    priority_rank: int = Field(alias="priorityRank")
    priority_tier: str = Field(alias="priorityTier")
    impact_level: str = Field(alias="impactLevel")
    recommended_next_action: str = Field(alias="recommendedNextAction")
    blocking_reason_summary: str = Field(alias="blockingReasonSummary")
    owner_surface: str = Field(alias="ownerSurface")
    remediation_surface: str = Field(alias="remediationSurface")


class AdminOpsLaunchCockpit(_AdminOpsStatusModel):
    contract: str = "admin_ops_launch_cockpit_v1"
    status: str = "unavailable"
    last_checked_at: Optional[str] = Field(default=None, alias="lastCheckedAt")
    message: str = ""
    read_only: bool = Field(default=True, alias="readOnly")
    advisory_only: bool = Field(default=True, alias="advisoryOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    public_launch_approved: bool = Field(default=False, alias="publicLaunchApproved")
    public_launch_no_go: bool = Field(default=True, alias="publicLaunchNoGo")
    live_enforcement: bool = Field(default=False, alias="liveEnforcement")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    approval_required: bool = Field(default=True, alias="approvalRequired")
    summary_counts: Dict[str, int] = Field(default_factory=dict, alias="summaryCounts")
    unsafe_action_states: Dict[str, bool] = Field(default_factory=dict, alias="unsafeActionStates")
    domains: List[AdminOpsCockpitDomain] = Field(default_factory=list)
    recommended_maintenance_queue: List[AdminOpsCockpitMaintenanceQueueItem] = Field(
        default_factory=list,
        alias="recommendedMaintenanceQueue",
    )
    blockers: List[AdminOpsCockpitBlocker] = Field(default_factory=list)
    safe_next_actions: List[str] = Field(default_factory=list, alias="safeNextActions")
    limitations: List[str] = Field(default_factory=list)
    priority_summary: Dict[str, int] = Field(default_factory=dict, alias="prioritySummary")


class AdminScannerUniverseReadinessResponse(_AdminOpsStatusModel):
    contract_version: str = Field(default="scanner_universe_operator_readiness_v1", alias="contractVersion")
    status: str
    scanner_universe_status: Optional[str] = Field(default=None, alias="scannerUniverseStatus")
    market: str = "cn"
    profile: str = "cn_preopen_v1"
    freshness_state: str = Field(default="unknown", alias="freshnessState")
    last_updated_at: Optional[str] = Field(default=None, alias="lastUpdatedAt")
    universe_size: int = Field(default=0, alias="universeSize")
    affected_product_surfaces: List[str] = Field(default_factory=list, alias="affectedProductSurfaces")
    next_operator_action: str = Field(default="", alias="nextOperatorAction")
    scanner_universe_readiness: Dict[str, Any] = Field(default_factory=dict, alias="scannerUniverseReadiness")
    candidate_generation_state: Optional[str] = Field(default=None, alias="candidateGenerationState")
    candidate_generation_blockers: List[str] = Field(default_factory=list, alias="candidateGenerationBlockers")
    read_only: bool = Field(default=True, alias="readOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    mutation_enabled: bool = Field(default=False, alias="mutationEnabled")
    provider_calls_enabled: bool = Field(default=False, alias="providerCallsEnabled")
    consumer_visible: bool = Field(default=False, alias="consumerVisible")


class AdminScannerUniverseRefreshResponse(_AdminOpsStatusModel):
    contract_version: str = Field(default="scanner_universe_operator_action_v1", alias="contractVersion")
    status: str
    action_status: str = Field(alias="actionStatus")
    market: str = "cn"
    profile: str = "cn_preopen_v1"
    refresh_executed: bool = Field(default=False, alias="refreshExecuted")
    mutation_enabled: bool = Field(default=False, alias="mutationEnabled")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    provider_calls_enabled: bool = Field(default=False, alias="providerCallsEnabled")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    next_operator_action: str = Field(default="", alias="nextOperatorAction")
    before: Dict[str, Any] = Field(default_factory=dict)
    after: Dict[str, Any] = Field(default_factory=dict)


class AdminBuildProvenance(_AdminOpsStatusModel):
    contract: str = "admin_build_provenance_v1"
    read_only: bool = Field(default=True, alias="readOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    consumer_visible: bool = Field(default=False, alias="consumerVisible")
    backend_git_sha: Optional[str] = Field(default=None, alias="backendGitSha")
    backend_branch: Optional[str] = Field(default=None, alias="backendBranch")
    backend_commit_timestamp: Optional[str] = Field(default=None, alias="backendCommitTimestamp")
    backend_runtime_started_at: Optional[str] = Field(default=None, alias="backendRuntimeStartedAt")
    frontend_main_asset_filename: Optional[str] = Field(default=None, alias="frontendMainAssetFilename")
    frontend_main_asset_hash: Optional[str] = Field(default=None, alias="frontendMainAssetHash")
    frontend_asset_manifest_hash: Optional[str] = Field(default=None, alias="frontendAssetManifestHash")
    frontend_asset_manifest_source: Optional[str] = Field(default=None, alias="frontendAssetManifestSource")
    frontend_static_build_timestamp: Optional[str] = Field(default=None, alias="frontendStaticBuildTimestamp")
    static_asset_mode: str = Field(default="unknown", alias="staticAssetMode")
    static_asset_root_provenance: str = Field(default="unknown", alias="staticAssetRootProvenance")
    static_asset_root_label: Optional[str] = Field(default=None, alias="staticAssetRootLabel")
    static_asset_root_exists: bool = Field(default=False, alias="staticAssetRootExists")
    static_index_present: bool = Field(default=False, alias="staticIndexPresent")
    freshness_status: str = Field(default="unknown", alias="freshnessStatus")
    comparison_basis: Optional[str] = Field(default=None, alias="comparisonBasis")
    stale: Optional[bool] = None
    reason_codes: List[str] = Field(default_factory=list, alias="reasonCodes")


class AdminOpsStatusResponse(_AdminOpsStatusModel):
    generated_at: str = Field(alias="generatedAt")
    read_only: bool = Field(default=True, alias="readOnly")
    no_external_calls: bool = Field(default=True, alias="noExternalCalls")
    live_enforcement: bool = Field(default=False, alias="liveEnforcement")
    runtime_behavior_changed: bool = Field(default=False, alias="runtimeBehaviorChanged")
    consumer_visible: bool = Field(default=False, alias="consumerVisible")
    advisory_vs_enforcement: AdminOpsAdvisoryVsEnforcement = Field(alias="advisoryVsEnforcement")
    provider_status_summary: AdminOpsStatusSection = Field(alias="providerStatusSummary")
    quota_cost_advisory_status_summary: AdminOpsStatusSection = Field(alias="quotaCostAdvisoryStatusSummary")
    storage_readiness_summary: AdminOpsStatusSection = Field(alias="storageReadinessSummary")
    task_queue_status_summary: AdminOpsStatusSection = Field(alias="taskQueueStatusSummary")
    admin_log_evidence_summary: AdminOpsStatusSection = Field(alias="adminLogEvidenceSummary")
    runtime_log_sink_summary: AdminOpsStatusSection = Field(alias="runtimeLogSinkSummary")
    retention_policy_status: AdminOpsStatusSection = Field(alias="retentionPolicyStatus")
    execution_log_retention_risk: AdminOpsStatusSection = Field(alias="executionLogRetentionRisk")
    db_size_risk: AdminOpsStatusSection = Field(alias="dbSizeRisk")
    admin_role_assignment_status: AdminOpsStatusSection = Field(alias="adminRoleAssignmentStatus")
    durable_task_backlog_status: AdminOpsStatusSection = Field(alias="durableTaskBacklogStatus")
    recommended_maintenance_actions: List[str] = Field(default_factory=list, alias="recommendedMaintenanceActions")
    build_provenance: AdminBuildProvenance = Field(alias="buildProvenance")
    launch_cockpit: AdminOpsLaunchCockpit = Field(alias="launchCockpit")
    metadata: Dict[str, Any] = Field(default_factory=dict)
